src/semgrep_agent/main.py

```diff
@@ -239,10 +243,18 @@ def main(
             meta.head_ref,
             semgrep.get_semgrepignore(sapp.scan.ignore_patterns),
             sapp.is_configured,
+            timeout=(timeout if timeout > 0 else None),
         )
     except SemgrepError as error:
         print_sh_error_info(error.stdout, error.stderr, error.command, error.exit_code)
         _handle_error(error.stderr, error.exit_code, sapp)
+    except sh.TimeoutException as error:
+        click.secho(
+            f"Semgrep took longer than {timeout} seconds to run; canceling this run",
```

Comment:
What next steps can we recommend to the user? Where do I go to change this timeout, investigate why it was slow, or otherwise debug?